addons/payment_authorize/models/payment_transaction.py

```diff
@@ -96,18 +96,42 @@ def _send_refund_request(self, amount_to_refund=None, create_refund_transaction=
                 create_refund_transaction=create_refund_transaction,
             )
 
+        authorize_API = AuthorizeAPI(self.acquirer_id)
+        tx_details = authorize_API.get_transaction_details(self.acquirer_reference)
+
+        # Avoid creating tx for validation operations.
+        create_refund_transaction = (tx_details and (not tx_details.get('err_code'))
+                                     and self.operation != 'validation')
+
         refund_tx = super()._send_refund_request(
-            amount_to_refund=amount_to_refund, create_refund_transaction=False
+            amount_to_refund=amount_to_refund, create_refund_transaction=create_refund_transaction
         )
 
-        authorize_API = AuthorizeAPI(self.acquirer_id)
-        rounded_amount = round(self.amount, self.currency_id.decimal_places)
-        res_content = authorize_API.refund(self.acquirer_reference, rounded_amount)
-        _logger.info(
-            "refund request response for transaction with reference %s:\n%s",
-            self.reference, pprint.pformat(res_content)
-        )
-        self._handle_notification_data('authorize', {'response': res_content})
+        if tx_details and tx_details.get('err_code'):
```

Comment:
You check here if the `tx_details` exists, but not in the `elif`. ;)

Reply:
You're right!, I fixed it :)